src_Core/RISCY_OOO/procs/RV64G_OOO/CommitStage.bsv

```diff
@@ -1379,6 +1422,25 @@ module mkCommitStage#(CommitInput inIfc)(CommitStage);
         );
     endrule
 
+`ifdef KONATA
+    rule doPrintCommitKONATA;
+        for(Integer i = 0; i < valueof(SupSize); i = i + 1) begin
+            if(printCommits[i][0] matches tagged Valid .u_id) begin
+                $display("KCommit print");
+                $display("KONATAE\t%0d\t%0d\t0\tC", cur_cycle, u_id);
+                $display("KONATAR\t%0d\t%0d\t%0d\t0", cur_cycle, u_id, u_id);
+                $fflush;
+            end
+        end
+    endrule
+
+    rule doMakePrintCommitInvalid;
+        for(Integer i = 0; i < valueof(SupSize); i = i + 1) begin
+            printCommits[i][0] <= Invalid;
```

Comment:
Hmm, this works, but a DWire might be simpler? It looks like the behaviour you want is just to get Invalid if it wasn't written to, and otherwise get the value written? DWire should do that without the need for this extra rule.

Reply:
I would like to have the content of the Ehr for the next cycle not for the current cycle. This is because I need to keep the content around in order to display the Commit stage in Konata. I _think_ I need an Ehr for that, but I might miss something and a DWire does the job.

Reply:
Quite right!